src/components/breadcrumbs/breadcrumbs.tsx

```diff
@@ -0,0 +1,217 @@
+ * laid out before the current page that the user is looking at.
+ * They could also be _steps_ which the user has gone through, which perhaps have no
+ * hierarchical relation with each other, but has eventually led the user "here".
+ *
+ * :::note
+ * - Where the user currently is, is always the last step of the breadcrumb.
+ * - A breadcrumbs never shows where users can go after this place.
+ * It only illustrates where user has been before ending up here.
+ * If the path that a user can take is not changing and if next steps are clear,
+ * you can use the [Progress flow component](#/component/limel-progress-flow) instead.
+ * :::
+ *
+ * Breadcrumbs are often placed horizontally before the main content of the current screen.
+ * @exampleComponent limel-example-breadcrumbs-links
+ * @exampleComponent limel-example-breadcrumbs-buttons
+ * @exampleComponent limel-example-breadcrumbs-icons
+ * @exampleComponent limel-example-breadcrumbs-divider
+ * @exampleComponent limel-example-breadcrumbs-icon-color
+ * @exampleComponent limel-example-breadcrumbs-styling
+ */
+
+@Component({
+    tag: 'limel-breadcrumbs',
+    shadow: true,
+    styleUrl: 'breadcrumbs.scss',
+})
+export class Breadcrumbs {
+    /**
+     * List of items in the breadcrumbs,
+     * each representing a step or a page.
+     */
+    @Prop()
+    public items: BreadcrumbsItem[];
+
+    /**
+     * The visual divider that separates items.
+     * It must be a single character such as `-` or `,`.
+     */
+    @Prop()
+    public divider: string = '›';
+
+    /**
+     * Used to attach the right tooltip to the right item
+     */
+    private tooltipId: string;
+
+    /**
+     * Fired when clicking on buttons (not links!)
+     * inside the breadcrumbs.
+     */
+    @Event()
+    public select: EventEmitter<BreadcrumbsItem>;
+
+    @Element()
+    private host: HTMLLimelBreadcrumbsElement;
+
+    private button: HTMLButtonElement;
+    private anchor: HTMLAnchorElement;
+
+    public render() {
+        return (
+            <ol
+                role="navigation"
+                aria-label="Breadcrumb"
+                style={{ '--limel-breadcrumbs-divider': `'${this.divider}'` }}
+            >
+                {this.renderSteps(this.items)}
+                {this.renderLastStep(this.items)}
+            </ol>
+        );
+    }
+
+    public componentWillRender() {
+        // Avoid polluting the "clickable registry" with unreferenced nodes
+        this.removeEnterClickable();
+    }
+
+    public componentDidRender() {
+        this.button = this.host.shadowRoot.querySelector('button');
+        this.anchor = this.host.shadowRoot.querySelector('a');
+
+        if (this.button) {
+            makeEnterClickable(this.button);
+        }
+
+        if (this.anchor) {
+            makeEnterClickable(this.anchor);
+        }
+    }
+
+    public disconnectedCallback() {
+        this.removeEnterClickable();
+    }
```

Comment:
@LucyChyzhova @LawrenceBorst we changed this section to:
```tsx
    public componentWillLoad() {
        makeEnterClickable(this.host);
    }

    public disconnectedCallback() {
        this.removeEnterClickable();
    }
```
which seems to produce the exact same effect. If you have any objections let us know. we don't exactly know why the code was written that way.

Reply:
@Kiarokh (and whoever the rest of "we" is, @Befkadu1 perhaps? 😊) It seems there was some cleanup missed when this was done. The `removeEnterClickable` still tried to remove the listeners from the anchor or button, which meant we had a leak here. But yes, not running two DOM queries and then setting up new event listeners on _every_ re-render is much better, so thanks for fixing that! 🙏 ❤️

I've opened #2748, which fixes the leak. Feel free to review! 🙏

Reply:
Thanks Adrian. I don't know anything about these lifecycle stuff and what leaks where or how to fix it. I hope someone else can look into these issues.